apis/crd/v1alpha1/cloudentityselector_webhook.go

```diff
@@ -218,27 +227,102 @@ func (r *CloudEntitySelector) validateMatchSections() error {
-							len(strings.TrimSpace(vmMatch.MatchName)) != 0 {
-							return fmt.Errorf("vpc matchName with either vm matchID" +
-								" or vm matchName is not supported, use vpc matchID instead" +
-								" of vpc matchName")
+			if m.VpcMatch != nil && len(strings.TrimSpace(m.VpcMatch.MatchName)) != 0 {
+				for _, vmMatch := range m.VMMatch {
+					if len(strings.TrimSpace(vmMatch.MatchID)) != 0 ||
+						len(strings.TrimSpace(vmMatch.MatchName)) != 0 {
+						return fmt.Errorf(errorMsgUnsupportedVPCMatchName02)
+					}
+				}
+				if m.Agented {
+					return fmt.Errorf(errorMsgUnsupportedAgented)
+				}
+			}
+		}
+	}
+
+	// Block ambiguous match combinations as it can result in unpredictable behavior w.r.t agented configuration.
+	if err := r.validateMatchCombinations(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// validateMatchCombinations function validates VMSelector to find if it conflicts with other VMSelectors.
+// Block same VPC ID configuration in two VMSelectors with only vpcMatch section.
+// Block same VM ID configuration in any two VMSelectors, same is not applicable for VM Name as VM Name need not be unique.
+// Block same combination of VPC ID and VM ID configuration in any two VMSelectors.
+// Block same combination of VPC ID and VM Name configuration in any two VMSelectors.
+// Block same VM Name configuration in any two VMSelectors with only VMMatch section, when used along with VPCMatch, it is allowed.
+func (r *CloudEntitySelector) validateMatchCombinations() error {
+	// vpcIDOnlyMatch map - VPC ID as key for selector with only vpcMatch matchID.
+	// vmIDOnlyMatch map - VM ID as key for selector with only vmMatch matchID.
+	// vmNameOnlyMatch map - VM Name as key for selector with only vmMatch matchName.
+	// vmIDWithVpcMatch map - VM ID and VPC ID as key for selector with vpcMatch matchID and vmMatch matchID.
+	// vmNameWithVpcMatch map - VM Name and VPC ID as key for selector with vpcMatch matchID and vmMatch matchName.
+	vpcIDOnlyMatch := make(map[string]struct{})
+	vmIDOnlyMatch := make(map[string]struct{})
+	vmNameOnlyMatch := make(map[string]struct{})
+	vmIDWithVpcMatch := make(map[string]struct{})
+	vmNameWithVpcMatch := make(map[string]struct{})
+	exists := struct{}{}
+
+	for _, selector := range r.Spec.VMSelector {
+		if selector.VpcMatch != nil {
+			if selector.VpcMatch.MatchID != "" {
+				if len(selector.VMMatch) == 0 {
+					if _, found := vpcIDOnlyMatch[selector.VpcMatch.MatchID]; found {
+						return fmt.Errorf("%s, %v", errorMsgSameVPCMatchID, selector.VpcMatch.MatchID)
+					}
+					vpcIDOnlyMatch[selector.VpcMatch.MatchID] = exists
+				} else {
+					for _, n := range selector.VMMatch {
+						if n.MatchID != "" {
+							index := n.MatchID + "/" + selector.VpcMatch.MatchID
+							if _, found := vmIDWithVpcMatch[index]; found {
+								return fmt.Errorf("%s, vpcMatch matchID %v, vmMatch matchID %v",
+									errorMsgSameVMAndVPCMatch, selector.VpcMatch.MatchID, n.MatchID)
```

Comment:
please verify the message with real setup to check for formatting and space.

Reply:
Will do and update here.

Reply:
Verified the error msg's are correctly formatted with a kind setup.